pkg/zonemgr/zone_manager.go

```diff
@@ -28,52 +28,26 @@ func NewZoneManager() *ZoneManager {
 }
 
 func (zoneMgr *ZoneManager) init() {
+	domain := os.Getenv("DOMAIN")
+	nameServerIP := os.Getenv("NAME_SERVER_IP")
 
-	fmt.Println("================ inside ZoneManager.init ===============")
+	zoneMgr.zoneFileCache = NewZoneFile(nameServerIP, domain)
+	zoneMgr.zoneFileCache.init()
 
-	//todo fetch custom details
-	clusterName := "cluster"
-	zoneFileName := "/zones/db.secdns"
-
-	userAdminEmail := ""
-	userSubdomain := "domain.com"
-	userNameserverName := ""
-	userNameserverIP := ""
-
-	zoneMgr.zoneFileCopy = NewZoneFile(clusterName, userAdminEmail,
-		userNameserverName, userNameserverIP, userSubdomain)
-
-	zoneMgr.zoneFileCopy.init()
-
-	//fmt.Println("================ zoneMgr.zoneFileCopy.content ===============\n", zoneMgr.zoneFileCopy.content)
-
-	zoneMgr.zoneFileDns = NewDnsZoneFile(zoneFileName)
-	// todo return zoneFileDns.writeFile(zoneFileCopy.content) // is it necessary to create an empty zone file on load??
+	zoneMgr.zoneFileDns = NewDnsZoneFile(zoneFileName + zoneMgr.zoneFileCache.domain)
 }
 
 func (zoneMgr *ZoneManager) UpdateZone(namespacedName k8stypes.NamespacedName, interfaces []v1.VirtualMachineInstanceNetworkInterface) (err error) {
-
-	//fmt.Println("================ inside ZoneManager.UpdateZone ===============")
-	//fmt.Println("================ namespacedName: ===============", namespacedName)
-	//fmt.Println("================ interfaces: ===============", interfaces)
-
-	isUpdated, err := zoneMgr.zoneFileCopy.updateVMIRecords(namespacedName.Name, namespacedName.Namespace, interfaces)
-	if err != nil {
-		return err
+	if namespacedName.Name == "" {
+		return errors.New("VM name in empty")
 	}
-
-	//fmt.Println("================ zoneMgr.zoneFileCopy.content: ===============\n", zoneMgr.zoneFileCopy.content)
-	//fmt.Println("================ zoneMgr.zoneFileDns: ===============\n", zoneMgr.zoneFileDns)
-
+	if namespacedName.Namespace == "" {
+		return errors.New("VM interface is empty")
+	}
+	isUpdated := zoneMgr.zoneFileCache.updateVMIRecords(namespacedName.Name, namespacedName.Namespace, interfaces)
```

Comment:
please consider passing the namespacedName to the function, the function can then have both fields with less parameters (as you already have it structured)

Reply:
done